src/transformers/models/t5/modeling_t5.py

```diff
@@ -1372,6 +1375,19 @@ def get_encoder(self):
     def get_decoder(self):
         return self.decoder
 
+    def _resize_token_embeddings(self, new_num_tokens):
+        old_input_embeddings = self.get_input_embeddings()
+        new_input_embeddings = self._get_resized_embeddings(old_input_embeddings, new_num_tokens)
+        self.set_input_embeddings(new_input_embeddings)
+
+        # if word embeddings are not tied, make sure that lm head is resized as well
+        if not self.config.tie_word_embeddings:
```

Comment:
Actually, I thought about this a bit more and I think we should be moving this `if` case to `modeling_utils.py`.
Since, `config.tie_word_embeddings` can be used by any model every model should have this if case. *E.g.*, the following code snippet currently fails:

```python
#!/usr/bin/env python3
from transformers import BertForMaskedLM

# model = BertForMaskedLM.from_pretrained("bert-base-cased") <- this would not fail
model = BertForMaskedLM.from_pretrained("bert-base-cased", tie_word_embeddings=False)
model.resize_token_embeddings(10)

model.save_pretrained("./")
BertForMaskedLM.from_pretrained("./")
```

even though it should be possible IMO. We can easily solve this by moving this code:

```
if not self.config.tie_word_embeddings and self.get_output_embeddings() is not None
            old_output_embeddings = self.get_output_embeddings()
            new_output_embeddings = self._get_resized_embeddings(old_output_embeddings, new_num_tokens)
            self.set_output_embeddings(new_output_embeddings)
```

into the `modeling_utils.py`s `_resize_token_embeddings` function. 

What do you think @sgugger @LysandreJik ?

Reply:
That should indeed work, great idea!

Reply:
Yes, good idea!